src/commands/postDeploy/getLinuxDetectorError.ts

```diff
@@ -1,83 +1,95 @@
-
-        const criticalErrorTable: detectorTable | undefined = findTableByRowValue(datasets, 'Critical');
-        const failureCountTable: detectorTable | undefined = findTableByColumnName(datasets, ColumnName.failureCount);
-
-        // if the criticalError table or failureCounts aren't defined, we can't display anything to the user
-        if (!criticalErrorTable || !failureCountTable) {
+    const rawApplicationLog: string = getValuesByColumnName(context, insightLogTable, ColumnName.value);
+    const insightDataset: detectorDataset[] = <detectorDataset[]>JSON.parse(rawApplicationLog);
+
+    let insightTable: detectorTable;
+    let detectorTimestamp: RegExpMatchArray | null;
+
+    const appInsightTable: detectorTable | undefined = findTableByName(insightDataset, 'application/insight');
+    if (appInsightTable) {
+        insightTable = appInsightTable;
+        context.telemetry.properties.insight = 'app';
+        detectorTimestamp = getValuesByColumnName(context, appInsightTable, ColumnName.dataName).match(timestampFormat);
+    } else {
+        // if there are no app insights, defer to the Docker container
+        const dockerInsightTable: detectorTable | undefined = findTableByName(insightDataset, 'docker/insight');
+        if (!dockerInsightTable) {
             return undefined;
         }
 
-        const errorMessages: string | undefined = getValuesByColumnName(context, criticalErrorTable, ColumnName.message);
-        const failureCount: string | undefined = getValuesByColumnName(context, failureCountTable, ColumnName.failureCount);
+        insightTable = dockerInsightTable;
+        context.telemetry.properties.insight = 'docker';
+        detectorTimestamp = getValuesByColumnName(context, dockerInsightTable, ColumnName.dataName).match(timestampFormat);
+    }
+
+    const deployTimestamp: RegExpMatchArray | null = deployResultTime.toISOString().match(timestampFormat);
+    if (!detectorTimestamp || !deployTimestamp || !validateTimestamp(context, detectorTimestamp[0], deployTimestamp[0])) {
+        return undefined;
+    }
+
+    const criticalError: string = localize('insightError', '"{0}" reported a critical error:', node.root.client.siteName);
 
-        context.telemetry.properties.errorMessages = JSON.stringify(errorMessages);
-        context.telemetry.properties.failureCount = failureCount;
+    if (getValuesByColumnName(context, insightTable, ColumnName.status) === 'Critical') {
+        const insightError: string = getValuesByColumnName(context, insightTable, ColumnName.dataValue);
+        context.telemetry.properties.errorMessages = JSON.stringify(insightError);
 
-        return `Critical insights found for "${siteName}": ${failureCount} as of ${new Date(timestamp).toLocaleString()}. Critical insight preview: "${errorMessages}". Click here to access "App Service Diagnostics" for recommendations.`;
+        return `${criticalError} ${insightError}".`;
```

Comment:
`insightError` should be added to the string inside of `localize` just like you do for `node.root.client.siteName` above.

There could be cases where languages read right-to-left instead of left-to-right and thus `insightError` would go first.